CMake/SiloFindPython.cmake

```diff
@@ -58,26 +58,31 @@
 ###
 
 
-if(DEFINED SILO_PYTHON_DIR)
+if(SILO_PYTHON_DIR)
     # help CMake find the specified python
-    set(Python_ROOT ${SILO_PYTHON_DIR})
+    set(Python3_ROOT_DIR ${SILO_PYTHON_DIR})
+    if(WIN32)
+        set(Python3_EXECUTABLE ${SILO_PYTHON_DIR}/python.exe)
+    else()
+        set(Python3_EXECUTABLE ${SILO_PYTHON_DIR}/bin/python3)
+    endif()
 endif()
 
-find_package(Python COMPONENTS Interpreter Development)
+find_package(Python3 COMPONENTS Interpreter Development)
```

Comment:
Does this mean we now require python3? I'd prefer not to have to *require* it if we can avoid that.

Reply:
Yes. I thought since Python 2 is no longer supported, that we should move to the new version for new releases.

Reply:
Yeah, Python 2 is no longer supported but a lot of computing centers still have it as the default and a lot of our customers I think still use it. @cyrus do you have thoughts on this?

Reply:
I think Python 2 is the default in a lot of computing centers and I think a lot of our customers still use Python 2. @cyrush do you have thoughts on this?

Reply:
I can remove the python3 changes if you want, no problem.

Reply:
Lets leave at python2 for now. With autotools, I know how to override to get python3 instead. I dunno how to do that with CMake.

Reply:
I reverted the changes that forced python3. Kept mods that help CMake find the python specified by SILO_PYTHON_DIR if provided.  Is preferring python3 in this case okay?